test/data_base.cpp

```diff
@@ -1135,4 +1145,395 @@ BOOST_AUTO_TEST_CASE(data_base__confirm__already_candidated___success)
+    BOOST_REQUIRE(header_result.metadata.validated);
+    const auto state = instance.blocks().get(header_result.hash()).state() & block_state::valid;
+    BOOST_REQUIRE(state != block_state::valid);
+}
+
+// catalog block
+
+//// BOOST_AUTO_TEST_CASE(data_base__catalog__enabled__success)
+//// {
+////     create_directory(DIRECTORY);
+////     bc::database::settings settings;
+////     settings.directory = DIRECTORY;
+////     settings.index_addresses = false;
+////     settings.flush_writes = false;
+////     settings.file_growth_rate = 42;
+////     settings.block_table_buckets = 42;
+////     settings.transaction_table_buckets = 42;
+////     settings.address_table_buckets = 42;
+
+////     data_base_accessor instance(settings);
+
+////     const auto bc_settings = bc::system::settings(config::settings::mainnet);
+////     BOOST_REQUIRE(instance.create(bc_settings.genesis_block));
+
+////     const auto block1 = read_block(MAINNET_BLOCK1);
+////     store_block_transactions(instance, block1, 1);
+
+////     BOOST_REQUIRE_EQUAL(instance.push_header(block1.header(), 1, 100), error::success);
+////     BOOST_REQUIRE_EQUAL(instance.candidate(block1), error::success);
+////     BOOST_REQUIRE_EQUAL(instance.update(block1, 1u), error::success);
+
+////     // setup ends
+
+////     BOOST_REQUIRE_EQUAL(instance.catalog(block1), error::success);
+
+////     // When address_database::catalog is implemented, enable this test
+////     BOOST_REQUIRE(instance.addresses().get(block1.transactions()[0].outputs()[0].address().hash()));
+//// }
+
+// catalog transactions
+
+//// BOOST_AUTO_TEST_CASE(data_base__catalog2__enabled__success)
+//// {
+////     create_directory(DIRECTORY);
+////     bc::database::settings settings;
+////     settings.directory = DIRECTORY;
+////     settings.index_addresses = false;
+////     settings.flush_writes = false;
+////     settings.file_growth_rate = 42;
+////     settings.block_table_buckets = 42;
+////     settings.transaction_table_buckets = 42;
+////     settings.address_table_buckets = 42;
+
+////     data_base_accessor instance(settings);
+
+////     const auto bc_settings = bc::system::settings(config::settings::mainnet);
+////     BOOST_REQUIRE(instance.create(bc_settings.genesis_block));
+
+////     const auto block1 = read_block(MAINNET_BLOCK1);
+////     store_block_transactions(instance, block1, 1);
+
+////     // setup ends
+
+////     BOOST_REQUIRE_EQUAL(instance.catalog(block1.transactions()[0]), error::success);
+
+////     // When address_database::catalog is implemented, enable this test
+////     BOOST_REQUIRE(instance.addresses().get(block1.transactions()[0].outputs()[0].address().hash()));
+//// }
+
+/// reorganize headers
+
+BOOST_AUTO_TEST_CASE(data_base__reorganize__pop_and_push__success)
+{
+    create_directory(DIRECTORY);
+    bc::database::settings settings;
+    settings.directory = DIRECTORY;
+    settings.index_addresses = false;
```

Comment:
This has recently changed, will require a rebase.

Reply:
Rebased.